packages/graphql-server-lambda/src/lambdaApollo.ts

```diff
@@ -0,0 +1,229 @@
+import * as lambda from 'aws-lambda';
+import * as graphql from 'graphql';
+import * as url from 'url';
+import { GraphQLOptions, runQuery } from 'graphql-server-core';
+import * as GraphiQL from 'graphql-server-module-graphiql';
+
+export interface LambdaGraphQLOptionsFunction {
+  (event: any, context: lambda.Context): GraphQLOptions | Promise<GraphQLOptions>;
+}
+
+// Design principles:
+// - there is just one way allowed: POST request with JSON body. Nothing else.
+// - simple, fast and secure
+//
+
+export interface LambdaHandler {
+  (event: any, context: lambda.Context, callback: lambda.Callback): void;
+}
+
+export interface IHeaders {
+    [header: string]: string | number;
+}
+
+export function graphqlLambda( options: GraphQLOptions | LambdaGraphQLOptionsFunction, headers?: IHeaders ): LambdaHandler {
+  if (!options) {
+    throw new Error('Apollo Server requires options.');
+  }
+  if (!headers) {
+    headers = {};
```

Comment:
I agree, we should eliminate the headers options and stick with one argument. Will make this change.